homeassistant/components/esphome/__init__.py

```diff
@@ -832,113 +842,80 @@ async def async_added_to_hass(self) -> None:
                 self._entry_data.signal_component_static_info_updated(
```

Comment:
Turns out we can subscribe by component key and key to reduce the number of callbacks here

Will need to update signal_component_static_info_updated above to include the keep and the dispatch can loop through and dispatch